compute_tools/src/http/middleware/authorize.rs

```diff
@@ -64,11 +64,26 @@ impl AsyncAuthorizeRequest<Body> for Authorize {
                 Err(e) => return Err(JsonResponse::error(StatusCode::UNAUTHORIZED, e)),
             };
 
-            if data.claims.compute_id != compute_id {
-                return Err(JsonResponse::error(
-                    StatusCode::UNAUTHORIZED,
-                    "invalid compute ID in authorization token claims",
-                ));
+            // If the scope if not [`ComputeClaimsScope::Admin`], then we must
+            // validate the compute_id.
+            if data
+                .claims
+                .scope
+                .is_some_and(|s| s != ComputeClaimsScope::Admin)
+            {
+                let Some(ref claimed_compute_id) = data.claims.compute_id else {
+                    return Err(JsonResponse::error(
+                        StatusCode::BAD_REQUEST,
+                        "missing compute_id in token claims",
```

Comment:
Afaik we don't propagate errors to caller, just yielding "invalid token" in this case. However, as this service is internal-facing only, fine to do that

Reply:
The reason for the message is because the control plane will log it.